variables.tf

```diff
@@ -1,4 +1,182 @@
+
+variable "availability_zone" {
+  type        = string
+  default     = null
+  description = "Optional parameter to place Amazon Redshift cluster instances in a specific availability zone. If left empty, will place randomly"
+}
+
+variable "preferred_maintenance_window" {
+  type        = string
+  default     = null
+  description = "Weekly time range during which system maintenance can occur, in UTC. Format: ddd:hh24:mi-ddd:hh24:mi"
+}
+
+variable "cluster_parameters" {
+  type = list(object({
+    name  = string
+    value = string
+  }))
+  default     = []
+  description = "List of Redshift parameters to apply"
+}
+
+variable "automated_snapshot_retention_period" {
+  type        = number
+  default     = 1
+  description = "The number of days that automated snapshots are retained. If the value is 0, automated snapshots are disabled"
+}
+
+variable "port" {
+  type        = number
+  default     = 5439
+  description = "The port number on which the cluster accepts incoming connections"
+}
+
+variable "engine_version" {
+  type        = string
+  default     = "1.0"
+  description = "The version of the Amazon Redshift engine to use."
+}
+
+variable "nodes" {
+  type        = number
+  default     = 1
+  description = "The number of compute nodes in the cluster. This parameter is required when the ClusterType parameter is specified as multi-node."
+}
+
+variable "publicly_accessible" {
+  type        = bool
+  default     = true
+  description = "If true, the cluster can be accessed from a public network."
+}
+
+variable "encrypted" {
+  type        = bool
+  description = "Specifies whether the cluster is encrypted at rest."
+  default     = false
+}
+
+variable "enhanced_vpc_routing" {
+  type        = bool
+  description = "If true , enhanced VPC routing is enabled."
+  default     = false
+}
+
+variable "kms_key_arn" {
+  type        = string
+  description = "The ARN for the KMS encryption key. When specifying `kms_key_arn`, `encrypted` needs to be set to `true`"
+  default     = null
+}
+
+variable "elastic_ip" {
+  type        = string
+  default     = null
+  description = "The Elastic IP (EIP) address for the cluster"
+}
+
+variable "skip_final_snapshot" {
+  type        = bool
+  default     = true
+  description = "Determines whether a final snapshot of the cluster is created before Amazon Redshift deletes the cluster."
+}
+
+variable "final_snapshot_identifier" {
+  type        = string
+  default     = null
+  description = "The identifier of the final snapshot that is to be created immediately before deleting the cluster. If this parameter is provided, `skip_final_snapshot` must be `false`"
+}
+
+variable "snapshot_identifier" {
+  type        = string
+  default     = null
+  description = "The name of the snapshot from which to create the new cluster."
+}
+
+variable "snapshot_cluster_identifier" {
+  type        = string
+  default     = null
+  description = "The name of the cluster the source snapshot was created from."
```

Comment:
I'm a bit confused about the name of this variable and the description.
is this the snapshot to create a new cluster from ?

Reply:
I think it's based on the upstream documentation https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/redshift_cluster#snapshot_cluster_identifier